### ISSUE TITLE
Does not run on Windows ARM devices

### ISSUE THREAD

**Issue by gslance** [owlbear-rodeo/kenku-fm#123]
The application does not run on Windows for ARM devices, such as the new Surface Pro 11+.

1. Have an ARM device running Windows
2. Download Windows executable from website (MacOS and Linux versions won't work anyway)
3. Install it
4. Open Kenku.FM as you would normally (from desktop or start menu)
5. No effect, nothing comes up. Administator rights have no change.

It would be nice to expect this to also work on devices running Windows for ARM. 

Running:
 - OS: Windows 11 ARM64
 - Version: Kenku.FM v1.4.3

The compatibility/emulation tool on the Surface Pro 11 (called Prism) is unable to emulate it. Many other software running x64/x86 work in this emulated way but not Kenku.FM.

**Comment by KERERCVASD** [comment]
try this
https://mega.co.nz/#!B31G2LaZ!uBBOCp9hLC7bq9kP8NC6s4DanQJJZuoFRr5FI_jkeic

you may need to install the c compiler

**Comment by gslance** [comment]
It seems that file is flagged as malware/virus. Can you provide some more information on it? What is changed to that version of this software?

**Comment by DemoBytom** [comment]
I'm hitting the same issue. I've gotten a Surface Pro 11 to use as a mobile DM tool, I've tried installing Kenku FM from the installer, but it doesn't let me do that. I downloaded the installer off the [site](https://www.kenku.fm/) and it crashes during the installation process.
I saw that there are versions for Linux ARM, so I'd hope Windows for ARM would be possible as well.

**Comment by gslance** [comment]
> I'm hitting the same issue. I've gotten a Surface Pro 11 to use as a mobile DM tool, I've tried installing Kenku FM from the installer, but it doesn't let me do that. I downloaded the installer off the [site](https://www.kenku.fm/) and it crashes during the installation process. I saw that there are versions for Linux ARM, so I'd hope Windows for ARM would be possible as well.

The only way I managed to get it to work was by installing Kenku through WSL (which is being deprecated by Microsoft). It was a bit of a hassle, but it did work. I don't really recommend it unless you're in dire need to run it. The issue is that you need to do extra steps if you want it to play audio on the same device (through Discord unaffected), you'd need to configure your WSL to use pulseaudio.

[Here's a complete guide how to set up WSL and the audio drivers. ](https://research.wmz.ninja/articles/2017/11/setting-up-wsl-with-graphics-and-audio.html)

From that, you can install Kenku FM using your installed Linux terminal and then run the application (I just made a shortcut to the batch file on the desktop so it starts like a regular app).

Again, not really recommended and I sincerely hope the devs take some time to do this officially. I tried to make a build myself, but the electron/forge or its dependencies don't support Windows on ARM completely when I tried it at the start of September, last month. I'm not proficient in Electron so I couldn't continue.

**Comment by nthouliss** [comment]
We just published a [new version of experimental](https://github.com/owlbear-rodeo/kenku-fm/releases/tag/v1.6.1-experimental) which includes a Windows ARM version. This has been tested on a Windows ARM VM. An ARM version only really became possible in the last 4 weeks.

@DemoBytom The experimental version intentionally does not have streaming settings. Cracks and glitches can occur when Kenku gets busy but if you're running it from a laptop make sure it's not running on power-saving settings. One of the better ways to ensure this is to have the laptop plugged into power.

**Comment by DemoBytom** [comment]
@nthouliss you are right, plugging the laptop in immediately stops the crackling. I can see the processor getting slightly throttled as I unplug, even though I have high performance profile set in the power settings. I can also see some process(es?) being occasionally put in power saving mode in task manager, regardless of the machine being plugged in or not:

<img width="1336" height="530" alt="Image" src="https://github.com/user-attachments/assets/0cc60a05-79e4-40b2-a585-f124395ea6e4" />

I don't know if that is an issue at all. I'll play around with the settings and see how it goes, but for now the glitches seem to be an issue on my end :)

I am running it on 16GB RAM Surface Pro 11, if there's something in particular that would be helpful to get tested on the hardware instead the VM, just let me know.

The `v1.6.1-experimental` installed with no problems, and atm I see no issues with either local or discord-streamed playback.
I tried playing an .ogg track, with 10+ layered soundboard tracks on top of it, while streaming it to Discord, and it works :)
